ui_testing/testcases/basic_tests/test04_testunits.py

```diff
@@ -812,3 +812,102 @@ def test025_edit_approach_overview_button(self):
         self.assertEqual(self.base_selenium.get_url(), '{}testUnits'.format(self.base_selenium.url))
         self.base_selenium.LOGGER.info('clicking on Overview confirmed')
 
+    def test_028_changing_testunit_type_update_fields_accordingly(self):
+        """
+        New: Test unit: Type Approach: When I change type from edit mode, the values should changed according to this type that selected 
+        When I change type from edit mode, the values should changed according to this type that selected 
+        LIMS-3680
+        """
+
+        self.base_selenium.LOGGER.info('open random testunit')
+
+        testunit_record = self.test_unit_page.get_random_test_units_row()
+        self.test_unit_page.open_edit_page(row=testunit_record)
+
+        self.base_selenium.LOGGER.info('set the type to Quantitative')
+        self.test_unit_page.set_testunit_type(testunit_type='Quantitative')
+        self.test_unit_page.sleep_tiny()
+        self.base_selenium.LOGGER.info('set testunit type to Quantitative, fields should be displayed as the following')
+        self.assertTrue(self.test_unit_page.check_for_quantitative_fields())
+
+        self.base_selenium.LOGGER.info('set the type to Qualitative')
+        self.test_unit_page.set_testunit_type(testunit_type='Qualitative')
+        self.test_unit_page.sleep_tiny()
+        self.base_selenium.LOGGER.info('testunit type is qualitative, fields shown should be as follow')
+        self.assertTrue(self.test_unit_page.check_for_qualitative_fields())
+        
+
+        self.base_selenium.LOGGER.info('set the type to Quantitative MiBi')
+        self.test_unit_page.set_testunit_type(testunit_type='Quantitative MiBi')
+        self.test_unit_page.sleep_tiny()
+        self.base_selenium.LOGGER.info('testunit type is qualitative, fields shown should be as follow')
+        self.assertTrue(self.test_unit_page.check_for_quantitative_mibi_fields())
+
+    def test_029_allow_user_to_change_between_specification_and_quantification(self):
```

Comment:
Redesign this test case. split it into 3 test cases.
maybe using test_unit_api with specific `filter` helps u

Reply:
the testcase doesn't specify changing from specification to quantification or the opposite, so the case is implemented to cover that whatever the type it has, it can be changed to the other type normally

Reply:
okay

Reply:
we need to test the three cases each time we run the test file